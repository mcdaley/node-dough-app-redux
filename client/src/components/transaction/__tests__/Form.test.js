//-----------------------------------------------------------------------------
// client/src/components/transaction/__tests__/Form.test.js
//-----------------------------------------------------------------------------
import React            from 'react'
import {
  render,
  cleanup,
}                       from '@testing-library/react'

import TransactionForm  from '../Form'

/**
 * TransactionForm Tests
 */
describe('TransactionForm', () => {
  const handler       = jest.fn(e => e.preventDefault())
  afterEach(cleanup)
    
  it('Takes a snapshot', () => {
    const { asFragment } = render(
      <TransactionForm accountId='xxx' onSubmit={handler} />
    )
    
    expect(asFragment(
      <TransactionForm  accountId='xxx' onSubmit={handler} />)).toMatchSnapshot()
  })
})