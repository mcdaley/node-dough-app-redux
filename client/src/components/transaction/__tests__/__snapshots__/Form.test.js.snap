// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionForm Takes a snapshot 1`] = `
<DocumentFragment>
  <div>
    <form
      class=""
      novalidate=""
      style="margin-bottom: 1.0rem;"
    >
      <table
        class="table table-dark table-sm"
      >
        <tbody>
          <tr>
            <td
              style="width: 15%;"
            >
              <div
                class="form-group"
                style="margin-bottom: 0px;"
              >
                <div>
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container"
                    >
                      <input
                        class="form-control"
                        id="date"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </td>
            <td
              style="width: 25%;"
            >
              <div
                class="form-group"
                style="margin-bottom: 0px;"
              >
                <input
                  class="form-control"
                  id="txn-description"
                  name="description"
                  placeholder="Description"
                  type="text"
                  value=""
                />
                <div
                  class="invalid-feedback"
                  data-testid="description-error"
                />
              </div>
            </td>
            <td
              style="width: 15%;"
            >
              <div
                class="form-group"
                style="margin-bottom: 0px;"
              >
                <input
                  class="form-control"
                  id="txn-category"
                  name="category"
                  placeholder="Category"
                  type="text"
                  value=""
                />
                <div
                  class="invalid-feedback"
                  data-testid="category-error"
                />
              </div>
            </td>
            <td
              style="width: 15%;"
            >
              <div
                class="form-group"
                style="margin-bottom: 0px;"
              >
                <input
                  class="form-control"
                  id="txn-debit"
                  name="debit"
                  placeholder="Debit"
                  type="text"
                  value=""
                />
                <div
                  class="invalid-feedback"
                  data-testid="debit-error"
                />
              </div>
            </td>
            <td
              style="width: 15%;"
            >
              <div
                class="form-group"
                style="margin-bottom: 0px;"
              >
                <input
                  class="form-control"
                  id="txn-credit"
                  name="credit"
                  placeholder="Credit"
                  type="text"
                  value=""
                />
                <div
                  class="invalid-feedback"
                  data-testid="credit-error"
                />
              </div>
            </td>
            <td
              style="width: 15%;"
            >
              <button
                class="btn btn-primary btn-md"
                style="margin-left: 0.50rem; margin-right: 0.50rem;"
                type="submit"
              >
                Save
              </button>
              <button
                class="btn btn-secondary btn-md"
                type="button"
              >
                Clear
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </form>
  </div>
</DocumentFragment>
`;
